File: app/src/main/java/igrek/songbook/system/SoftKeyboardService.java
package igrek.songbook.system;

import android.content.Context;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

import javax.inject.Inject;

import igrek.songbook.dagger.DaggerIoc;

public class SoftKeyboardService {
	
	@Inject
	AppCompatActivity activity;
	
	private InputMethodManager imm;
	
	public SoftKeyboardService() {
		DaggerIoc.getFactoryComponent().inject(this);
		imm = (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
	}
	
	public void hideSoftKeyboard(View view) {
		if (imm != null) {
			imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
		}
	}
	
	public void hideSoftKeyboard() {
		//Find the currently focused view, so we can grab the correct window token from it.
		View view = activity.getCurrentFocus();
		//If no view currently has focus, create a new one, just so we can grab a window token from it
		if (view == null) {
			view = new View(activity);
		}
		hideSoftKeyboard(view);
	}
	
	public void showSoftKeyboard(View view) {
		if (imm != null) {
			imm.showSoftInput(view, 0);
		}
	}
}

File: app/src/main/java/igrek/songbook/system/SoftKeyboardService.kt
package igrek.songbook.system

import android.content.Context
import android.view.View
import android.view.inputmethod.InputMethodManager
import androidx.appcompat.app.AppCompatActivity
import igrek.songbook.info.logger.LoggerFactory
import igrek.songbook.inject.LazyExtractor
import igrek.songbook.inject.LazyInject
import igrek.songbook.inject.appFactory

class SoftKeyboardService(
        appCompatActivity: LazyInject<AppCompatActivity> = appFactory.appCompatActivity,
) {
    private val activity by LazyExtractor(appCompatActivity)

    private val imm: InputMethodManager? = activity.getSystemService(Context.INPUT_METHOD_SERVICE) as? InputMethodManager
    private val logger = LoggerFactory.logger

    fun hideSoftKeyboard(view: View?) {
        if (imm == null) {
            logger.error("no input method manager")
            return
        }
        if (view == null) {
            logger.error("view = null")
            return
        }
        imm.hideSoftInputFromWindow(view.windowToken, 0)
    }

    fun hideSoftKeyboard() {
        //Find the currently focused view, so we can grab the correct window token from it.
        var view = activity.currentFocus
        //If no view currently has focus, create a new one, just so we can grab a window token from it
        if (view == null) {
            view = View(activity)
        }
        hideSoftKeyboard(view)
    }

    fun showSoftKeyboard(view: View?) {
        imm?.showSoftInput(view, 0)
    }
}
